plugin/common/plugin.h

```diff
@@ -57,15 +58,16 @@ class BaseCreator : public IPluginCreator
 template <typename T>
 void write(char*& buffer, const T& val)
 {
-    *reinterpret_cast<T*>(buffer) = val;
+    std::memcpy(buffer, &t, sizeof(T));
```

Comment:
&val instead of &t?

Reply:
absolutely! I'm away from a machine with an nvida graphics card for some time, so I'll have to wait to verify everything (but I'll make this change for sure)